client.go

```diff
@@ -364,6 +364,48 @@ func (h hookLogger) Printf(s string, args ...interface{}) {
 	h.Info(fmt.Sprintf(s, args...))
 }
 
+// ContextLogger is an interface that provides methods for logging
+// with context. The methods accept a context.Context, a message
+// string and a variadic number of key-value pairs.
+type ContextLogger interface {
+	ErrorContext(ctx context.Context, msg string, keysAndValues ...interface{})
+	InfoContext(ctx context.Context, msg string, keysAndValues ...interface{})
+	DebugContext(ctx context.Context, msg string, keysAndValues ...interface{})
+	WarnContext(ctx context.Context, msg string, keysAndValues ...interface{})
+}
+
+// contextLogger adapts an ContextLogger to Logger for use by the existing hook functions
+// without changing the API.
+type contextLogger struct {
+	logger ContextLogger
```

Comment:
Please make logger public or embedded, so that it is possible to access the original logger in hooks.

Reply:
Provided for external use is 'ContextLogger' interface, and the role of 'contextLogger' struct is the same as that of 'hookLogger'.

Reply:
`hookLogger` allows you to access wrapped `Logger`. `contextLogger` does not.

Reply:
Thank you very much. It feels much smoother.

Reply:
So maybe just embed `ContextLogger`, the same how `hookLogger` is embedding `LeveledLogger`. Then it can be accessible if needed from hooks.

Reply:
Yes, I made changes